tools/otci/otci/otci.py

```diff
@@ -2148,13 +2148,150 @@ def set_domain_name(self, name: str):
+        # Done
+        #
+
+        status = ''
+        report_received = False
+        ret = False
+
+        for line in output:
+            if 'Received Link Metrics Management Response from' in line:
+                address = line.split(': ')[1].strip()
+                report_received = address == peer_addr
+            elif 'Status' in line:
+                status = line.split(':')[1].strip()
+
+        if report_received and status == 'Success':
+            ret = True
+
+        return ret
+
+    def __parse_linkmetrics_report(self, peer_addr: Union[str, Ip6Addr], output: List[str]) -> Dict[str, int]:
+        #
+        # Example output:
+        #
+        # Received Link Metrics Report from: fe80:0:0:0:3092:f334:1455:1ad2
+        #
+        # - PDU Counter: 2 (Count/Summation)
+        # - LQI: 76 (Exponential Moving Average)
+        # - Margin: 82 (dB) (Exponential Moving Average)
+        # - RSSI: -18 (dBm) (Exponential Moving Average)
+        # Done
+        #
+
+        results = {}
+        report_received = False
+
+        for line in output:
+            if 'Received Link Metrics Report' in line:
+                address = line.split(': ')[1].strip()
+                report_received = address == peer_addr
+            elif 'Received Link Metrics data in Enh Ack from neighbor' in line:
+                # If the Enhanced-ACK Based Probing is enabled, the CLI will output the following
+                # link metrics info after executing the `linkmetrics request` command. This case is
+                # used to skip these Enhanced-ACK related link metrics info.
+                #
+                # Received Link Metrics data in Enh Ack from neighbor, short address:0x3400 , extended address:c6a24d6514cf9178
+                # - LQI: 224 (Exponential Moving Average)
+                # - Margin: 0 (dB) (Exponential Moving Average)
+                #
+                # Received Link Metrics Report from: fe80:0:0:0:3092:f334:1455:1ad2
+                #
+                # - PDU Counter: 2 (Count/Summation)
+                # - LQI: 76 (Exponential Moving Average)
+                # - Margin: 82 (dB) (Exponential Moving Average)
+                # - RSSI: -18 (dBm) (Exponential Moving Average)
+                # Done
+                #
+                report_received = False
+
+            if not report_received:
+                continue
+
+            if '- LQI' in line:
+                results['lqi'] = self.__parse_line_numbers(line)[0]
+            elif '- Margin' in line:
+                results['margin'] = self.__parse_line_numbers(line)[0]
+            elif '- RSSI' in line:
+                results['rssi'] = self.__parse_line_numbers(line)[0]
+            elif '- PDU Counter' in line:
+                results['pdu_counter'] = self.__parse_line_numbers(line)[0]
+            else:
+                continue
+
+        return results
+
+    def __parse_line_numbers(self, line: str) -> List[int]:
```

Comment:
The method's name makes me think it's parsing the line number.

Reply:
I have updated the name to `__parse_numbers`